src/Shared/CertificateGeneration/CertificateManager.cs

```diff
@@ -323,6 +323,14 @@ protected virtual void PopulateCertificatesFromStore(X509Store store, List<X509C
         {
             try
             {
+                // If the user specified a non-existent directory, we don't want to be responsible
+                // for setting the permissions appropriately, so we'll bail.
+                var exportDir = Path.GetDirectoryName(path);
+                if (!string.IsNullOrEmpty(exportDir) && !Directory.Exists(exportDir))
+                {
+                    throw new InvalidOperationException($"The directory '{exportDir}' does not exist.");
```

Comment:
Should the error provide a friendly pointer that users should choose permissions carefully?

Reply:
"When creating it, think carefully about who should be able to read it?"